src/components/ProductCard/ProductCard.tsx

```diff
@@ -6,17 +7,38 @@ import {
   Stack,
   Typography,
 } from "@mui/material";
+import { useRef } from "react";
 import { Product } from "../../types/types";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
+import { addItem, removeItem } from "../../store/features/cartSlice";
+
+export const ProductCard = ({ product }: { product: Product }) => {
+  const productRef = useRef<HTMLDivElement | null>(null);
+  const cart = useAppSelector((state) => state.cart);
+  const dispatch = useAppDispatch();
+
+  const handleAddToCart = (item: Product) => {
```

Comment:
let's have a hook, that is called `useProductCart()`
It will have `handleAddToCart`, `handleRemoveFromCart` functions

this hook should be called inside the parent component of the ProductCart. Not inside of the ProductCart.

ProductCart should receive props:
- product
- onAddToCart
- onRemoveFromCart

The reason to have this logic passed through the props is that it will be easier to control.
For instance, you want to also show the toast notification on the page, when the cart is full, or there are no more products to be added to the cart. Then, you can modify a parent component of the cart, where there will be:

```
const productCart = useProductCart()
const handleAddToCart = ({ product }) => {
  if(product.qty > 20) { // example
   showSomeMessageInTheNotification()
  }
  productCart.handleAddToCart(product)
}

```

Reply:
Done, but not sure whether it is correct